### DISCUSSION TITLE
CPU usage showing 100% memory usage

### DISCUSSION THREAD

**Discussion by salmanAhmad143** [spatie/laravel-short-schedule#61] [Q&A]
Hi,

Thanks for creating this awesome library. I just starting to use this. My requirement was to run the command every 20 seconds. Everything was working fine but as soon as I moved it to the production server. Suddenly my CPU usage has been showing 100% memory usage. I don't know what's going on and how to handle this problem. 

`
**protected function shortSchedule(\Spatie\ShortSchedule\ShortSchedule $shortSchedule)
    {
        // this artisan command will run every 20 second
        $shortSchedule->command('check:connectionExpire')->everySeconds(20);   
    }**
   
`
**Below is the command that I had set up on my server crontab-**

`php /var/www/mySite/artisan short-schedule:run 1>> /dev/null 2>&1`

**Could you please suggest to me what I can do to overcome this problem?**

Awaiting your reply.

Regards,
Salman Ahmad

**Comment by freekmurze** [comment]
I can only say that I'm not seeing that problem on my end.

Should you find that the package is in error, feel free to submit a PR that resolves the problem.

**Reply by salmanAhmad143** [reply]
Hi, @freekmurze I am still struggling to find the solution. Every time when I run the short-schedule command on my digital ocean server. After a few minutes server has been crashing and it starts to show me 100% CPU usage. 

I read the documentation and you suggested to deal with memory leak need to use an extra parameter "**--lifetime**". Is this help me here to deal with this problem?

`php artisan short-schedule:run --lifetime=60`

Please suggest to me anyone some solution?

Regards,
Salman Ahmad

**Reply by piglet-NOT-pig** [reply]
I have the same problem. Have you solved it

**Reply by paulgotea** [reply]
Same problem here. As I saw, for example, if I set the command to run every 5 seconds, works fine at the beginning and then it just duplicates several times. Any solutions?

**Comment by GuljahanG** [comment]
How solve this?

**Reply by paulgotea** [reply]
Make sure you use -> withoutOverlapping()

**Comment by SoufAsd** [comment]
i had the same issue when using it in production server , i decided later to use a supervisor to run it

**Reply by firmanalams** [reply]
Are you solved this using supervisor?

**Reply by SoufAsd** [reply]
Yes like this :
```
..
 command=/usr/bin/php {your project folder path}/artisan short-schedule:run
 ..
```

**Reply by firmanalams** [reply]
can you share your supervisor config?
